src/main/java/com/example/bookstoreapp/services/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,114 @@
+        int totalItemCount = cart
+                .getItems()
+                .stream()
+                .mapToInt(CartItem::getQuantity)
+                .sum();
+        shoppingOrder.setTotalItemCount(totalItemCount);
+        List<OrderItem> orderItems = cart
+                .getItems()
+                .stream()
+                .map(this::convertToOrderItem).toList();
+        shoppingOrder.setOrderItems(orderItems);
+        Address address = extractAddress(orderRequest.getAddressId());
+        String addressStr = AppUtils.toString(address);
+        shoppingOrder.setAddress(addressStr);
+        ShoppingOrderEntity entity = shoppingOrderEntityRepository.save(shoppingOrder.toEntity());
+        List<OrderItemEntity> orderItemEntities = shoppingOrder
+                .getOrderItems()
+                .stream()
+                .map(e->{
+                    e.setOrderId(entity.getId());
+                    return e.toEntity(entity.getId());
+                })
+                .toList();
+        orderItemEntityRepository.saveAll(orderItemEntities);
+
+        log.info("order amount: {} placed: {}, with item count: {} from cart: {}",
+                shoppingOrder.getTotalAmount(),
+                shoppingOrder.getId(),
+                shoppingOrder.getTotalItemCount(),
+                cart.getCartId());
+
+        return shoppingOrder;
+    }
+    private OrderItem convertToOrderItem(CartItem cartItem){
+        OrderItem orderItem = new OrderItem();
+        orderItem.setId(IdGenerator.getLongId());
+        orderItem.setCatalogItemId(cartItem.getId());
+        orderItem.setUnitPrice(cartItem.getUnitPrice());
+        orderItem.setQuantity(cartItem.getQuantity());
+        orderItem.setTotal(cartItem.getTotal());
+        orderItem.setCatalogItemId(cartItem.getId());
+        orderItem.setCatalogItemId(cartItem.getId());
+        orderItem.setCartId(cartItem.getCartId());
+        orderItem.setPurchasedOn(new Date());
+        orderItem.setStatusCode(StatusCode.ORDER_PLACED);
+        return orderItem;
+    }
+    private Address extractAddress(Long addressId) {
+        List<Address> addressList = addressService.findUserAddress(userContextService.getUserId());
+        if (addressList.isEmpty()) {
+            throw new AppRuntimeException("please add an address for order");
+        }
+        Address address = addressList
+                .stream()
+                .filter(element -> element.getId().equals(addressId))
+                .findFirst()
+                .orElseThrow(() -> new AppRuntimeException("Address not found"));
+        return address;
+        // Delete cart items associated with the order
+        List<CartItem> cartItems=deleteCartItems(cartItemService.doCheckout(userContextService.getUserId()));
+    }
+    private void deleteCartItems(List<CartItem> cartItems) {
+        for (CartItem cartItem : cartItems) {
+            CartItemEntityRepository.delete(cartItem);
```

Comment:
Here we are accessing delete method using interface name. Is it getting compiled?

We should take help of variable cartItemService and ask him to delete.

OrderService can talk to order repository.
But if OrderService need to talk others then go through service layer and don't use repository layer call like you did above.

So I will suggest something like below

cartItemService.delete(buyNow) and write its implementation.

So it will delete all items whose state is buy now for the logged in user.

So you have to use a jpa delete query whose equivalent sql query like:

Delete from cart where cart_id =? And state=?

State is weight is buyNow
CartId can be found from user Context service.